lib/internal/modules/esm/get_format.js

```diff
@@ -158,7 +158,7 @@ function getFileProtocolModuleFormat(url, context = { __proto__: null }, ignoreE
           if (!source) { return null; }
           const format = getFormatOfExtensionlessFile(url);
           if (format === 'module') {
-            return containsModuleSyntax(`${source}`, fileURLToPath(url)) ? 'module' : 'commonjs';
+            return containsModuleSyntax(`${source}`, fileURLToPath(url), url) ? 'module' : 'commonjs';
```

Comment:
Why don't you just pass url as a parameter, and do the fileURLToPath in C++?

Reply:
This is not unified and can be a non URL (or not even a path) at times, so it's just easier to control it from JS. Also it's not very performance critical from the ESM path (it's already a slow path if you don't do explicit type and want Node.js to compile it twice as different formats).